app/models/concerns/authorizable.rb

```diff
@@ -16,7 +16,7 @@ module Authorizable
       if user.nil?
         self.where('1=0')
       elsif user.admin?
-        self.scoped
+        self.where(nil)
```

Comment:
This looks like the most likely candidate to me, since Katello seems to be calling `System.deletable.where(...)`, which is a scope that calls `authorized(:destroy_content_hosts)`